Source/WebCore/dom/DocumentMarkerController.cpp

```diff
@@ -644,9 +644,21 @@ void DocumentMarkerController::shiftMarkers(Node& node, unsigned startOffset, in
         // FIXME: No obvious reason this should be iOS-specific. Remove the #if at some point.
         auto targetStartOffset = clampTo<unsigned>(static_cast<int>(marker.startOffset()) + delta);
         auto targetEndOffset = clampTo<unsigned>(static_cast<int>(marker.endOffset()) + delta);
-        if (targetStartOffset >= node.length() || targetEndOffset <= 0) {
-            list->remove(i);
-            continue;
+
+        // If `delta < 0`, then text is being removed, and the actual start offset of the text being
+        // removed is `startOffset - delta`.
+        auto startOffsetOfAffectedRange = static_cast<int>(startOffset);
+        if (delta < 0)
+            startOffsetOfAffectedRange += delta;
+
+        if (marker.endOffset() > clampTo<unsigned>(startOffsetOfAffectedRange)) {
```

Comment:
Shouldn't this comparison be against `marker.startOffset()`?

Reply:
I don’t think; this is just testing if the marker overlaps the removed text range; so marker end > removed text range start